rpc/daemon.go

```diff
@@ -0,0 +1,49 @@
+package rpc
+
+import (
+	"fmt"
+	"net/http"
+	"syscall"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+
+	"github.com/athanorlabs/atomic-swap/cliutil"
+	"github.com/athanorlabs/atomic-swap/common"
+	"github.com/athanorlabs/atomic-swap/net"
+)
+
+// DaemonService handles RPC requests for swapd version, administration and (in the future) status requests.
+type DaemonService struct {
+	server *Server
+	pb     ProtocolBackend
+}
+
+// NewDaemonService ...
+func NewDaemonService(server *Server, pb ProtocolBackend) *DaemonService {
+	return &DaemonService{
+		server,
+		pb,
+	}
+}
+
+// Shutdown swapd
+func (s *DaemonService) Shutdown(req *http.Request, _ *any, _ *any) error {
+	return syscall.Kill(syscall.Getpid(), syscall.SIGINT)
```

Comment:
Replaced `server.Stop()` with a signal (handled here https://github.com/AthanorLabs/atomic-swap/blob/master/cmd/swapd/signal_handler.go) because it was shutting down the RPC server before sending a response to the shutdown request, and caused a EOF error on the client.

Maybe it would be better to manage to call the cancel function of the outermost context directly instead of relying on the signal handler.

Reply:
I thought about it some more, and we can just cancel the RPC server's context, not the outermost context. The code changes are relatively minor: [PR on top of this PR's branch](https://github.com/stubbrn/atomic-swap/pull/1/files)